File: backend/app/routers/customers.py
"""Customer HTTP endpoints."""
from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.schemas.customer import CustomerCreate, CustomerOut, CustomerUpdate
from app.services import customer as service

router = APIRouter(prefix="/customers", tags=["customers"])


@router.get("", response_model=list[CustomerOut])
def list_customers(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
    return service.list_customers(db, skip, limit)


@router.post("", response_model=CustomerOut, status_code=status.HTTP_201_CREATED)
def create_customer(data: CustomerCreate, db: Session = Depends(get_db)):
    return service.create_customer(db, data)


@router.get("/{customer_id}", response_model=CustomerOut)
def get_customer(customer_id: int, db: Session = Depends(get_db)):
    return service.get_customer(db, customer_id)


@router.put("/{customer_id}", response_model=CustomerOut)
def update_customer(customer_id: int, data: CustomerUpdate, db: Session = Depends(get_db)):
    return service.update_customer(db, customer_id, data)


@router.delete("/{customer_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_customer(customer_id: int, db: Session = Depends(get_db)):
    service.delete_customer(db, customer_id)

File: frontend/src/api/customers.js
import client from './client';

export async function listCustomers() {
  const { data } = await client.get('/customers');
  return data;
}

export async function getCustomer(id) {
  const { data } = await client.get(`/customers/${id}`);
  return data;
}

export async function createCustomer(payload) {
  const { data } = await client.post('/customers', payload);
  return data;
}

export async function updateCustomer(id, payload) {
  const { data } = await client.put(`/customers/${id}`, payload);
  return data;
}

export async function deleteCustomer(id) {
  await client.delete(`/customers/${id}`);
}
